Add find to netrunnerdb-cards

muyjohno opened the pull request:
Replicates NRDB-style searching of cards, e.g. `!find s:sentry s!tracer p>3` will find sentries that can't be broken by Mimic or Gingerbread.

(Suggestions or feedback appreciated)

Commit:
Add find to netrunnerdb-cards

Replicates NRDB-style searching of cards, e.g. `!find s:sentry s!tracer p>3` will find sentries that can't be broken by Mimic or Gingerbread

dodgepong commented on scripts/hubot-netrunnerdb-cards.coffee:
```diff
@@ -239,6 +239,36 @@ compareCards = (card1, card2) ->
 		return 0
 
+cardMatches = (card, cond) ->
+	switch cond.op
```
Does this need a default `else` case?

muyjohno replied:
It's a good question. In theory no, because invalid conditions have already been stripped out. I guess it should default to some truthy value just in case though (i.e. it will ignore invalid conditions). I'll make the tweak and update.

dodgepong commented on scripts/hubot-netrunnerdb-cards.coffee:
```diff
@@ -314,3 +344,31 @@ module.exports = (robot) ->
+	robot.hear /^!find (.*)/, (res) ->
+		conditions = []
+		for part in res.match[1].toLowerCase().split(" ")
+			if out = part.match(/([a-z]+)(:|=|<|>|!)([\w]+)/)
```
Should the `([a-z]+)` group be limited only to valid filter parameters (`etsfxpondcaiuy`)? Also, does it need to allow multiple characters for a parameter? It appears that search filter keys are always single characters, so I don't think a `+` is necessary here, unless I'm missing something.

With that limitation, the conditionals below could probably be slightly simplified.

muyjohno replied:
Yeah, good catch. I think originally I was going to allow longer keys (e.g. "influence") but I decided to copy NRDB one-char codes in the end. Will update.

Commit:
Default unrecognised find conditions to true

(i.e. ignore them)

Commit:
Refactor validation

dodgepong commented on scripts/hubot-netrunnerdb-cards.coffee:
```diff
@@ -314,3 +343,27 @@ module.exports = (robot) ->
+
+	robot.hear /^!find (.*)/, (res) ->
+		conditions = []
+		for part in res.match[1].toLowerCase().split(" ")
```
It looks like there may be an issue if you want to search for things like multi-word subtypes. On NRDB I can search for `s:"code gate"` to return Code Gates, but it looks like this code would split that into two separate filters. :/

Commit:
Allow multi-word terms within quotes

e.g. `!find s:"code gate" x:"secretly spend"` will return Bullfrog and Clairvoyant Monitor
